### DISCUSSION TITLE
Can this library meet my use case?

### DISCUSSION THREAD

**Discussion by msmcconnell** [pelicanmapping/osgearth#2282] [Support Q&A]
I am looking for guidance from the community on if osgearth can meet my use case. I want to provide my system with a collection of high resolution map tiles from satellite imagery with elevation information. At runtime I need to be able to render nearby tiles from a specified 3d position and orientation. The orientation is 3d and therefore may contain the horizon and be non-normal to the nearest tile. Since the altitude can vary widely it's important for performance that the level of detail rendered be constrained to what is actually visible given the rendering resolution. Nothing would actually be displayed, this is an entirely internal computational process where the rendered view will be leveraged later (as an image) in my system's pipeline. 

Is this something osgearth can support? Are there any concerns that jump out to the reader?

**Comment by XenonofArcticus** [comment]
So basically you want osgEarth to cull the necessary tiles and compute LOD,
but not actually render them?

**Reply by msmcconnell** [reply]
Yes to the first two points. However, I do need it to render to an image just not display on screen.

**Comment by gwaldron** [comment]
osgEarth will render a 3D view from any orientation, so that part has you covered. By default it loads and renders only visible tiles and only tiles at an "appropriate" resolution for the camera parameters, so you should be good there as well.

As for off-screen rendering: osgEarth doesn't do this natively; but it is built on top of the 3D rendering library OpenSceneGraph which itself has mechanisms for this.

I'd recommend starting an evaluation by getting osgEarth up and running, trying some of the examples, and loading your data. You will quickly get a sense of whether it will suit your needs.

**Reply by msmcconnell** [reply]
Thanks for the info. If I use osgEarth to set the orientation and tile parameters can I still access the underlying OpenSceneGraph object to perform the off-screen rendering?

**Reply by gwaldron** [reply]
Yes, the osgEarth globe is represented by an OSG node called the `MapNode` that lives in the scene graph.

**Reply by msmcconnell** [reply]
Great thanks! Seems like it's worth digging into it further then.
